code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -94,6 +94,28 @@
 	if(!alien_weeds || alien_weeds.secreting)
 		return SECRETE_RESIN_FAIL
 
+	var/list/blacklist_caste = list(XENO_CASTE_HIVELORD)
+	var/list/whitelist_build = list(
+		/datum/resin_construction/resin_turf/wall,
+		/datum/resin_construction/resin_turf/wall/thick,
+		/datum/resin_construction/resin_turf/wall/queen,
+		/datum/resin_construction/resin_turf/membrane,
+		/datum/resin_construction/resin_turf/membrane/thick,
+		/datum/resin_construction/resin_turf/membrane/queen,
+		/datum/resin_construction/resin_turf/wall/reflective,
+		/datum/resin_construction/resin_obj/door,
+		/datum/resin_construction/resin_obj/door/thick,
+		/datum/resin_construction/resin_obj/door/queen,
+	)
```

Comment:
Ideally, these lists should be taken out either as datums or as internal variables, but this is not worth doing.